charts/datadog/templates/rbac.yaml

```diff
@@ -109,6 +109,17 @@ rules:
   resources: ["secrets"]
   verbs: ["get"]
 {{- end }}
+{{- if .Values.datadog.helmCheck.enabled}}
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  - configmaps
+  verbs:
+  - get
+  - list
+  - watch
+{{- end }}
```

Comment:
Instead of defining this on `agent-clusterchecks-rbac.yaml` I think the best solution might be to create a dedicated file similar to `kube-state-metrics-core-rbac.yaml` like you suggested.